lib/DXIL/DxilOperations.cpp

```diff
@@ -3421,12 +3423,11 @@ void OP::UpdateCache(OpCodeClass opClass, Type *Ty, llvm::Function *F) {
 }
 
 Function *OP::GetOpFunc(OpCode opCode, Type *pOverloadType) {
-  DXASSERT(0 <= (unsigned)opCode && opCode < OpCode::NumOpCodes,
-           "otherwise caller passed OOB OpCode");
-  assert(0 <= (unsigned)opCode && opCode < OpCode::NumOpCodes);
```

Comment:
There are a few other places in this file where we replicate this assert (which itself is duplicated). Can we remove those cases? These asserts seem silly since they're basically asserting "we don't have undefined behavior".

Reply:
Removed.

Reply:
I'm not sure if this is what Chris meant. In this location we have `DXASSERT` and `assert`, which are entirely redundant. In this case, the robust responses below make both asserts less useful, but in many cases where asserts got removed entirely including those I commented on above, that is not the case.